app/extras/model_locator.rb

```diff
@@ -5,6 +5,8 @@ def locate
       resource_class.find id_param
     elsif key_param = params[:"#{model}_key"]
       resource_class.find_by_key key_param
+    elsif model == :user
```

Comment:
Yep, so I want this route to work:
```
/u/:username
```
as well as this one:
```
/u/:key/:username
```

Because, when we construct the urls for @mentions, we don't necessarily have that user loaded into the record store, and I'm a bit loathe to try to, server side, figure out what users are mentioned in a window of activity. The thing we do have is the username though, so I can take this:
```
<p>I'd like to hear @loomiobot 's opinion!</p>
```
and turn it into this:
```
<p>I'd like to hear <a href="/u/loomiobot">@loomiobot</a> 's opinion</p>
```
but I can't turn it into this:
```
<p>I'd like to hear <a href="/u/botkey/loomiobot">@loomiobot</a> 's opinion</p>
```
because I don't know the key for the user.